eogrow/pipelines/split_grid.py

```diff
@@ -0,0 +1,184 @@
+"""Implements a pipeline that creates a finer grid and splits EOPatches accordingly."""
+import logging
+from collections import defaultdict
+from typing import Dict, List, Literal, Tuple, Union
+
+import fs
+import geopandas as gpd
+from pydantic import Field
+
+from eolearn.core import EONode, EOWorkflow, FeatureType, LoadTask, OverwritePermission, SaveTask
+from sentinelhub import BBox
+
+from ..core.area.batch import BatchAreaManager
+from ..core.area.utm import UtmZoneAreaManager
+from ..core.pipeline import Pipeline
+from ..tasks.spatial import SpatialSliceTask
+from ..utils.fs import LocalFile
+from ..utils.grid import split_bbox
+from ..utils.types import Feature, FeatureSpec
+
+LOGGER = logging.getLogger(__name__)
+
+NamedBBox = Tuple[str, BBox]
+
+
+class SplitGridPipeline(Pipeline):
+    """Pipeline that creates a finer grid and splits EOPatches accordingly.
+
+    The new grid is output as a geopackage file, which can be used with a `CustomAreaManager`.
+    """
+
+    class Schema(Pipeline.Schema):
+        input_folder_key: str = Field(
+            description="A storage manager key pointing to the folder from where data will be loaded."
```

Comment:
"A storage manager key pointing to the folder where the data will be loaded."*

Reply:
Done, added "from" to the end